xfid.go

```diff
@@ -938,16 +942,18 @@ func xfideventread(x *Xfid, w *Window) {
 	fc.Count = uint32(n)
 	fc.Data = w.events[:n]
 	x.respond(&fc, nil)
-	nn := len(w.events)
-	copy(w.events[0:], w.events[n:])
 
-	w.events = w.events[0 : nn-n]
+	w.events = w.events[n:]
 }
 
 func xfidindexread(x *Xfid) {
 	// log.Println("xfidindexread", x)
 	// defer log.Println("done xfidindexread")
 
+	// BUG(fhs): This is broken when the client is doing a sequential
```

Comment:
Like what `win` does sometimes? And this is why `win` gets confused over a slow link when a command run from it creates windows? Is this behaviour in Acme too? That would explain something that I see from time to time.

Reply:
This bug exists in acme too. I'm not sure about the issue with `win`. I was using this program to simulate slow read with a small buffer:
```Go
package main

import (
	"io"
	"os"
	"time"

	"9fans.net/go/plan9"
	"9fans.net/go/plan9/client"
)

func main() {
	fsys, err := client.MountService("acme")
	if err != nil {
		panic(err)
	}
	index, err := fsys.Open("index", plan9.OREAD)
	if err != nil {
		panic(err)
	}
	defer index.Close()

	var buf [12]byte

	for {
		n, err := index.Read(buf[:])
		if err == io.EOF {
			break
		}
		if err != nil {
			panic(err)
		}
		os.Stdout.Write(buf[:n])
		time.Sleep(time.Second)
	}
}
```